corehq/apps/reports/standard/cases/case_data.py

```diff
@@ -555,6 +560,207 @@ def close_case_view(request, domain, case_id):
+                case_object = touched_cases[touched_id].case
+                actions = list(touched_cases[touched_id].actions)
+                if touched_id == case.case_id:
+                    case_actions['current'] = actions
+                elif touched_id not in delete_cases:
+                    if touched_id not in case_actions:
+                        case_actions[case_object.name] = actions
+                    if const.CASE_ACTION_CREATE in actions and touched_id != case.case_id:
+                        get_forms_for_deletion_from_case(case_object, subcase_count + 1)
+                    if const.CASE_ACTION_CLOSE in actions:
+                        reopened_cases[touched_id] = form_id
+                        case_names[touched_id] = case_object.name
+                    if any(action in actions for action in update_actions):
+                        if touched_id not in affected_cases:
+                            affected_cases[touched_id] = {}
+                        affected_cases[touched_id][form_id] = ', '.join(actions)
+                        case_names[touched_id] = case_object.name
+            cases[case.case_id][form_id] = case_actions
+
+    case_instance = safely_get_case(request, domain, case_id)
+    subcase_count = 0
+    try:
+        get_forms_for_deletion_from_case(case_instance, subcase_count)
+    except TooManyCases:
+        messages.error(request, _("Deleting this case would delete too many related cases. "
+                                  "Please delete some of this cases' subcases before attempting"
+                                  "to delete this case."))
+        return {'redirect': True}
+
+    def get_case_link(caseid):
+        url = reverse('case_data', args=[domain, caseid])
+        return mark_safe('<a href="{}"> {} </a>'.format(url, case_names[caseid]))  # case names are already escaped
+
+    def get_form_link(formid):
+        url = reverse('render_form_data', args=[domain, formid])
+        return mark_safe('<a href="{}"> {} </a>'.format(url, escape(form_names[formid])))
+
+    prepared_cases = {}
+    for case in cases:
+        prepared_forms = {}
+        for form in cases[case]:
+            form_link = get_form_link(form)
+            prepared_forms[form_link] = {}
+            for case_action in cases[case][form]:
+                prepared_forms[form_link][case_action] = ', '.join(cases[case][form][case_action])
+        prepared_cases[get_case_link(case)] = prepared_forms
+
+    prepared_reopened_cases = {}
+    for case in reopened_cases:
+        prepared_reopened_cases[get_case_link(case)] = get_form_link(reopened_cases[case])
+
+    prepared_affected_cases = {}
+    for case in affected_cases:
+        if case in delete_cases:
+            continue
+        prepared_forms = {}
+        for form in affected_cases[case]:
+            prepared_forms[get_form_link(form)] = affected_cases[case][form]
+        prepared_affected_cases[get_case_link(case)] = prepared_forms
```

Comment:
to tie in mentions of a data structure. perhaps something like
```
AffectedCase = namedtuple('AffectedCase', ['name', 'url', 'affected_forms'])
AffectedForm = namedtuple('AffectedForm', ['name, 'url', 'actions'])
```

and then in the template, you can call on the properties by iterating through a list of structured items rather than `key, value` pairs from `affected_cases.items`